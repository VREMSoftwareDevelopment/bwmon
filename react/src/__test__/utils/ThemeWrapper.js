import React from 'react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { create } from 'react-test-renderer';
import theme from './../../theme';

export const themeWrapper = (component) =>
    create(
        <StyledEngineProvider injectFirst>
            <ThemeProvider theme={theme}>{component}</ThemeProvider>
        </StyledEngineProvider>
    );
