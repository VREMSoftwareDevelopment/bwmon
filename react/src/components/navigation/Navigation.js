/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from 'mui-styles';
import { BottomNavigation, BottomNavigationAction } from '@mui/material';
import useNavigation from './UseNavigation';

const useStyles = makeStyles({});

const Navigation = ({ menu }) => {
    const classes = useStyles();
    const { index, setIndex } = useNavigation(menu);

    const handleChange = (event, newRouteIndex) => setIndex(newRouteIndex);

    return (
        <BottomNavigation value={index} onChange={handleChange} showLabels className={classes.root}>
            {menu.map((route, index) => (
                <BottomNavigationAction
                    data-testid={route.id}
                    id={route.id}
                    key={route.pathname}
                    value={index}
                    label={route.label}
                    icon={route.icon}
                    component={Link}
                    to={route.pathname}
                />
            ))}
        </BottomNavigation>
    );
};

export default Navigation;
