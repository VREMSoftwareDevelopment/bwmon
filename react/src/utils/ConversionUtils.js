/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import { DateTime } from 'luxon';

export const timeToDate = (time) => DateTime.fromSeconds(Number(time)).toFormat('MMM dd, yyyy HH:mm');
export const toMonth = (month) => DateTime.local(2012, month).toFormat('MMMM');
export const usageInGBytes = (value) => (Number(value) / 1000000).toFixed(3);
export const toPercent = (value) => Number(value).toFixed(1) + '%';
export const toIPv4 = (value) => (value >>> 24) + '.' + ((value >> 16) & 255) + '.' + ((value >> 8) & 255) + '.' + (value & 255);
export const fromIPv4 = (value) => value.split('.').reduce((value, octet) => (value << 8) + parseInt(octet, 10), 0) >>> 0;
