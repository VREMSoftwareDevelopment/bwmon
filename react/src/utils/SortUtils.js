/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

const isAscending = (orderBy, property, ascending) => (orderBy === property ? !ascending : false);

const comparator = (isAscending, orderBy) => {
    const ascending = (a, b, orderBy) => {
        if (a[orderBy] < b[orderBy]) return -1;
        if (a[orderBy] > b[orderBy]) return 1;
        return 0;
    };
    const descending = (a, b, orderBy) => {
        if (a[orderBy] > b[orderBy]) return -1;
        if (a[orderBy] < b[orderBy]) return 1;
        return 0;
    };
    return isAscending ? (a, b) => ascending(a, b, orderBy) : (a, b) => descending(a, b, orderBy);
};

const sort = (values, comparator) => [...values].sort(comparator);

export { comparator, isAscending, sort };
