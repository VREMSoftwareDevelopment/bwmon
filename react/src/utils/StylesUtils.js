import { makeStyles } from '@mui/styles';

const defaultStyles = makeStyles({});

const footerStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        margin: theme.spacing(2),
    },
}));

const graphStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        margin: theme.spacing(2),
        marginTop: 6,
    },
}));

const headerStyles = makeStyles({
    title: {
        flexGrow: 1,
    },
});

const messageStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    },
}));

const paginationStyles = makeStyles((_theme) => ({
    root: {
        flexShrink: 0,
    },
}));

export { defaultStyles, footerStyles, graphStyles, headerStyles, messageStyles, paginationStyles };
