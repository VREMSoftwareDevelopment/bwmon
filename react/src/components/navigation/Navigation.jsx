/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { BottomNavigation, BottomNavigationAction } from '@mui/material';
import useNavigation from './UseNavigation';
import { defaultStyles } from '../../utils/StylesUtils';

const Navigation = ({ menu }) => {
    const classes = defaultStyles();
    const { index, setIndex } = useNavigation(menu);

    const handleChange = (event, newRouteIndex) => setIndex(newRouteIndex);

    return (
        <BottomNavigation value={index} onChange={handleChange} showLabels className={classes.root}>
            {menu.map((route, index) => (
                <BottomNavigationAction
                    data-testid={route.id}
                    id={route.id}
                    key={route.pathname}
                    value={index}
                    label={route.label}
                    icon={route.icon}
                    component={Link}
                    to={route.pathname}
                />
            ))}
        </BottomNavigation>
    );
};

Navigation.propTypes = {
    menu: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            pathname: PropTypes.string.isRequired,
            label: PropTypes.string.isRequired,
            icon: PropTypes.element.isRequired,
        })
    ).isRequired,
};

export default Navigation;
