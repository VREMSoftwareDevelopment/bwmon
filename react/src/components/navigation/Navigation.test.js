/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { themeWrapper } from '../../__test__/utils/themeWrapper';
import Navigation from './Navigation';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
        pathname: '/pathname2',
    }),
}));

describe('Navigation', () => {
    const menu = [
        { id: 'id1', pathname: '/pathname1', label: 'label1', icon: 'icon1' },
        { id: 'id2', pathname: '/pathname2', label: 'label2', icon: 'icon2' },
        { id: 'id3', pathname: '/pathname3', label: 'label3', icon: 'icon3' },
    ];

    test('renders correctly', () => {
        const tree = themeWrapper(
            <BrowserRouter>
                <Navigation menu={menu} />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
