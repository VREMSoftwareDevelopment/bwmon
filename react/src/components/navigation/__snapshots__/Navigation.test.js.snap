// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div
  className="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
>
  <a
    className="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
    href="/pathname1"
    id="id1"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    icon1
    <span
      className="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
    >
      label1
    </span>
  </a>
  <a
    className="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
    href="/pathname2"
    id="id2"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    icon2
    <span
      className="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
    >
      label2
    </span>
  </a>
  <a
    className="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
    href="/pathname3"
    id="id3"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    icon3
    <span
      className="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
    >
      label3
    </span>
  </a>
</div>
`;
